src/map/itemdb.cpp

```diff
@@ -1086,34 +1087,28 @@ s_item_combo *itemdb_combo_exists(uint32 combo_id) {
 	return item.get();
 }
 
-/**
-* Check if item group exists
-* @param group_id
-* @return NULL if not exist, or s_item_group_db *
-*/
-struct s_item_group_db *itemdb_group_exists(unsigned short group_id) {
-	return (struct s_item_group_db *)uidb_get(itemdb_group, group_id);
-}
-
 /**
  * Check if an item exists in a group
  * @param group_id: Item Group ID
  * @param nameid: Item to check for in group
  * @return True if item is in group, else false
  */
-bool itemdb_group_item_exists(unsigned short group_id, t_itemid nameid)
+bool itemdb_group_item_exists(uint16 group_id, t_itemid nameid)
 {
-	struct s_item_group_db *group = (struct s_item_group_db *)uidb_get(itemdb_group, group_id);
-	unsigned short i, j;
+	std::shared_ptr<s_item_group_db> group = itemdb_group.find(group_id);
 
-	if (!group)
+	if (group == nullptr || group->random.empty())
```

Comment:
```suggestion
	if (group == nullptr)
```
Not necessary since the for loop below will cancel early in this case.